downloadCurrentVersion.ts

```diff
@@ -3,73 +3,26 @@
 // This is used to download the correct binary version
 // as part of the prepublish step.
 
-import * as fs from 'fs';
-import { https } from 'follow-redirects';
-import MemoryStream from 'memorystream';
-import { keccak256 } from 'js-sha3';
+import downloader from './downloader';
 const pkg = require('./package.json');
 
-function getVersionList (cb) {
-  console.log('Retrieving available version list...');
-
-  const mem = new MemoryStream(null, { readable: false });
-  https.get('https://binaries.soliditylang.org/bin/list.json', function (response) {
-    if (response.statusCode !== 200) {
-      console.log('Error downloading file: ' + response.statusCode);
-      process.exit(1);
+async function download () {
+  try {
+    const list = JSON.parse(await downloader.getVersionList());
+    const wanted = pkg.version.match(/^(\d+\.\d+\.\d+)$/)[1];
+    const releaseFileName = list.releases[wanted];
+    const expectedFile = list.builds.filter(function (entry) { return entry.path === releaseFileName; })[0];
+    if (!expectedFile) {
+      throw new Error('Requested version not found. Version list is invalid or corrupted?');
     }
-    response.pipe(mem);
-    response.on('end', function () {
-      cb(mem.toString());
-    });
-  });
-}
-
-function downloadBinary (outputName, version, expectedHash) {
-  console.log('Downloading version', version);
-
-  // Remove if existing
-  if (fs.existsSync(outputName)) {
-    fs.unlinkSync(outputName);
-  }
-
-  process.on('SIGINT', function () {
-    console.log('Interrupted, removing file.');
-    fs.unlinkSync(outputName);
+    const expectedHash = expectedFile.keccak256;
+    await downloader.downloadBinary('soljson.js', releaseFileName, expectedHash);
+    process.exit();
```

Comment:
Do we need an explicit exit? Wouldn't it be better to let the script simply end?

Reply:
No we don't. I'll remove it.